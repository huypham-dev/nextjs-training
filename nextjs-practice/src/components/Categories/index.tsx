import { memo } from "react";

import { CategoryType } from "types";
import CategoryItem from "./Item";
import isEqual from "react-fast-compare";

type Props = {
  data: CategoryType[];
};

const Categories = ({ data }: Props) => (
  <div className="mx-auto max-w-6xl bg-white py-10 px-12 lg:px-24 shadow-xl mb-6">
    <h1 className="font-display-italic font-extrabold text-5xl md:text-6xl italic leading-none mb-12">
      Categories
    </h1>
    <div className="flex flex-wrap -mx-2">
      {data.map((item) => (
        <CategoryItem
          key={item.id}
          imageUrl={item.image}
          categoryName={item.name}
        />
      ))}
    </div>
  </div>
);

export default memo(Categories, isEqual);
