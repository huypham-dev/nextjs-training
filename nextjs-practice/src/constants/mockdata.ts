export const CATEGORIES = [
  {
    id: "1",
    name: "mobile",
    image:
      "https://images.prismic.io/nextjsexample/7853f863-6a12-4312-a6da-f930437f6d62_mobile.png",
  },
  {
    id: "2",
    name: "internet",
    image:
      "https://images.prismic.io/nextjsexample/cb71ce70-bb03-442b-99e8-a318255b81f3_3579187.png",
  },
];

export const ARTICLES = [
  {
    id: "1",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
  {
    id: "2",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
];
