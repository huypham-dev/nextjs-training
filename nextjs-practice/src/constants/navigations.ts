export const navigations = [
  { label: "Home", path: "/" },
  { label: "Blog", path: "/blog" },
  { label: "Contact", path: "/contact" },
];
