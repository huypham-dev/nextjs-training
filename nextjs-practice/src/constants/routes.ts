export const ROUTES = {
  home: {
    path: "/",
  },
  blog: {
    path: "/blog",
    withId: (id: string) => `/blog/${id}`,
  },
  post: {
    path: "/post",
    withId: (id: string) => `/post/${id}`,
  },
  contact: {
    path: "/",
  },
};
