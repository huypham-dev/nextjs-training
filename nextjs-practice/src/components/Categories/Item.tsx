import { memo } from "react";

import Image from "next/image";
import LinkBase from "components/Link";
import { ROUTES } from "constants/routes";
import Typography from "components/common/Typography";

type Props = {
  imageUrl: string;
  categoryName: string;
};

const CategoryItem = ({ imageUrl, categoryName }: Props) => (
  <div className="w-full md:w-1/2 lg:w-1/3 px-6 p-2">
    <LinkBase
      className="block mb-8 text-center hover:opacity-70"
      href={ROUTES.blog.withId(categoryName)}
    >
      <Image src={imageUrl} alt={categoryName} width={100} height={100} />
    </LinkBase>
    <Typography className="text-xl text-center">
      {categoryName.toUpperCase()}
    </Typography>
  </div>
);

export default memo(CategoryItem);
