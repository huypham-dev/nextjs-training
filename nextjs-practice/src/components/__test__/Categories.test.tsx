import { mount, shallow } from "enzyme";
import { CATEGORIES } from "constants/mockData";
import Categories from "../Categories";
import Item from "../Categories/Item";

describe("Categories", () => {
  it("Snapshot", () => {
    const component = shallow(<Categories data={CATEGORIES} />);
    expect(component).toMatchSnapshot();
  });

  it("Render component correctly with props", () => {
    const component = mount(<Categories data={CATEGORIES} />);
    const title = component.find("h1");
    expect(title.text()).toBe("Categories");
    expect(component.find(Item).length).toBe(2);
  });
});
