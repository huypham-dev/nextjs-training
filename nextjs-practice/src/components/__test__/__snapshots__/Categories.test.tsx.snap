// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Snapshot 1`] = `
<div
  className="mx-auto max-w-6xl bg-white py-10 px-12 lg:px-24 shadow-xl mb-6"
>
  <Memo(Typography)
    className="font-display-italic font-extrabold text-5xl md:text-6xl italic leading-none mb-12"
    variant="h2"
  >
    Categories
  </Memo(Typography)>
  <div
    className="flex flex-wrap -mx-2"
  >
    <Memo(CategoryItem)
      categoryName="mobile"
      imageUrl="https://images.prismic.io/nextjsexample/7853f863-6a12-4312-a6da-f930437f6d62_mobile.png"
      key="1"
    />
    <Memo(CategoryItem)
      categoryName="internet"
      imageUrl="https://images.prismic.io/nextjsexample/cb71ce70-bb03-442b-99e8-a318255b81f3_3579187.png"
      key="2"
    />
  </div>
</div>
`;
